File: 백준/Silver/2839. 설탕 배달/설탕 배달.py
def sugar(n):
    result = n // 5 # 5개짜리 봉지 수 확인
    balance = n % 5 # 3으로 나눠지는지 확인할 봉지
    
    if balance == 0:
        return result
    
    while True:
        if balance % 3 == 0:
            result += balance // 3
            return result
        else:
            result -= 1
            balance += 5
            if balance > n:
                return -1
        
n = int(input())
memo = {}

print(sugar(n))

File: 백준/Silver/2839. 설탕 배달/설탕 배달.cc
#include <iostream>

using namespace std;

int sugar(int n) {
	int result = n / 5;
	int balance = n % 5;
	
	if(balance==0){
		return result;
	}
	
	while (true){
        if(balance % 3 == 0){
            result += balance / 3;
            return result;
        }
        else {
            result -= 1;
            balance += 5;
            if (balance > n){
	            return -1;
            }
        }
  }
}

int main() {
    int n;
    cin >> n;
    cout << sugar(n) << endl;
    return 0;
}
